src/com/palmergames/bukkit/towny/utils/SpawnUtil.java

```diff
@@ -0,0 +1,366 @@
+						throw new TownyException(String.format(TownySettings.getLangString("msg_err_x_spawn_disallowed_from_x"), spawnType.getTypeName(), TownySettings.getLangString("msg_enemy_areas")));
+					if (!inNation.hasAlly(playerNation) && !inNation.hasEnemy(playerNation) && disallowedZones.contains("neutral"))
+						throw new TownyException(String.format(TownySettings.getLangString("msg_err_x_spawn_disallowed_from_x"), spawnType.getTypeName(), TownySettings.getLangString("msg_neutral_towns")));
+				}
+			}
+		}
+
+		Double travelCost = 0.0;
+		String spawnPermission = null;
+		TownyEconomyObject payee = null;
+		// Figure out costs, payee and spawnPermmission slug for money.csv log.
+        switch (spawnType) {
+        case RESIDENT:        	
+        	// Taking whichever is smaller, the cost of the spawn price set by the town, or the cost set in the config (which is the maximum a town can set their spawncost to.)
+        	travelCost = Math.min(townSpawnPermission.getCost(town), townSpawnPermission.getCost());
+        	spawnPermission = String.format(spawnType.getTypeName() + " (%s)", townSpawnPermission);
+        	payee = town;
+        	break;        
+        case TOWN:
+			// Taking whichever is smaller, the cost of the spawn price set by the town, or the cost set in the config (which is the maximum a town can set their spawncost to.)
+			travelCost = Math.min(townSpawnPermission.getCost(town), townSpawnPermission.getCost());
+        	spawnPermission = String.format(spawnType.getTypeName() + " (%s)", townSpawnPermission);
+        	payee = town;
+        	break;        	
+        case NATION:
+            // Taking whichever is smaller, the cost of the spawn price set by the nation, or the cost set in the config (which is the maximum a nation can set their spawncost to.)
+            travelCost = Math.min(nationSpawnPermission.getCost(nation),nationSpawnPermission.getCost());
+        	spawnPermission = String.format(spawnType.getTypeName() + " (%s)", nationSpawnPermission);
+        	payee = nation;
+        	break;
+        default:
+        	break;
+        }
+
+		// Check if need/can pay.
+		try {
+			if ((!townyUniverse.getPermissionSource().has(player, PermissionNodes.TOWNY_COMMAND_TOWNYADMIN_TOWN_SPAWN_FREECHARGE.getNode())) && 
+					(travelCost > 0 && TownySettings.isUsingEconomy() && (resident.getHoldingBalance() < travelCost)))
+				throw new TownyException(notAffordMSG);
+		} catch (EconomyException ignored) {
+		}
+
+		// Used later to make sure the chunk we teleport to is loaded.
+		Chunk chunk = spawnLoc.getChunk();
+		
+		// Essentials tests.
+		boolean usingESS = plugin.isEssentials();
+
+		if (usingESS && !isTownyAdmin) {
+			try {
+				User essentialsUser = plugin.getEssentials().getUser(player);
+
+				// This jail check is specifically for essentials jails, not towny ones.
+				if (!essentialsUser.isJailed()) {
+
+					Teleport teleport = essentialsUser.getTeleport();
+					if (!chunk.isLoaded())
+						chunk.load();

```

Comment:
Do not force-load the chunk if essentials is being used. The reason being is that Essentials caters appropriately to servers using Paper. Thus, when a player is teleported, essentials will use Paper's async chunk loading to load the chunk and teleport them. However, `chunk.load()` will cause a synchronized chunk load even on Paper servers and negate the benefit of async loading.

Reply:
Alright, this code is from before spigot/paper existed.